### ISSUE TITLE
Use custom KRB5_CONFIG and turn off dns_canonicalize_hostname in keysystem

### ISSUE THREAD

**Issue by celskeggs** [sipb/homeworld#299]
We both want to:

- isolate ourselves from the choices that our users make about configuring their systems
- and make sure that dns_canonicalize_hostname (or at least rdns) is disabled for the sake of security and not trusting reverse DNS

as such, we should give knc a KRB5_CONFIG env variable and specify exactly what config it should use.

The main question is "what should this config look like", because it's possible that the user does want to do something with a nonstandard kerberos realm.

**Comment by krawthekrow** [comment]
If that's the case, I expect people to generally have different methods for identity verification. From my understanding of Hyades' architecture, the identity verification part is already somewhat isolated from the rest of the system -- maybe we should make it such that that component can be easily swapped out?

**Comment by celskeggs** [comment]
Yes, that's probably a better use of time than trying to make our krb5 configuration support arbitrary domains.
